src/android/NetworkManager.java

```diff
@@ -218,30 +207,25 @@ private void unregisterReceiver() {
     private void updateConnectionInfo(NetworkInfo info) {
         // send update to javascript "navigator.network.connection"
         // Jellybean sends its own info
-        JSONObject thisInfo = this.getConnectionInfo(info);
-        if(!thisInfo.equals(lastInfo))
+        String currentNetworkType = this.getTypeOfNetworkFallbackToTypeNoneIfNotConnected(info);
+        if(currentNetworkType.equals(this.lastTypeOfNetwork))
         {
```

Comment:
```suggestion
        if (currentNetworkType.equals(this.lastTypeOfNetwork)) {
```

Reply:
done. In the other parts of the NetworkManager class, there are also strange indentations, but I rather leave them untouched.